classes/curlmanager_security_helper.php

```diff
@@ -59,64 +59,64 @@ private function log_curl_http_requests(string $urlstring) : bool {
         $curlsecurityhelper = new curl_security_helper();
         $urlblocked = $curlsecurityhelper->url_is_blocked($urlstring);
 
-        $rootcodepath = '';
-        $trace = debug_backtrace();
-        $formattedbacktrace = format_backtrace(debug_backtrace(), true);
-        $lasttrace = count($trace) - 1;
-        if (isset($trace[$lasttrace]['file'])) {
-            $rootcodepath = $trace[$lasttrace]['file'];
-        }
-
-        $plugin = $this->getcomponentbycodepath($rootcodepath);
-        if ($plugin === false) {
-            $plugin = '';
-        }
+        // At Some Point TM this will need to be changed when this gets true unit tests.
+        // For now, let's stop it from busting core tests.
+        $log = get_config('tool_curlmanager', 'loggingenabled') && !PHPUNIT_TEST;
+        if ($log) {
+            $rootcodepath = '';
+            $trace = debug_backtrace();
+            $formattedbacktrace = format_backtrace(debug_backtrace(), true);
+            $lasttrace = count($trace) - 1;
+            if (isset($trace[$lasttrace]['file'])) {
+                $rootcodepath = $trace[$lasttrace]['file'];
+            }
 
-        // Suggest to deduplicate on host, plugin and codepath.
-        // Check if the host, plugin and codepath exists already.
-        // Add a new record if not exist.
-        // Otherwise update the reocrd with count+1 and timeupdated field.
-        $record = $DB->get_records('tool_curlmanager',
-            ['host' => $host, 'plugin' => $plugin],
-            '',
-            'id, count'
-        );
-
-        if (count($record) > 0) {
-            $record = current($record);
-            $data = new \stdClass();
-            $data->id = $record->id;
-            $data->count = $record->count + 1;
-            $data->codepath = $formattedbacktrace;
-            $data->urlallowed = $returnvalue['allowed'] ? 1 : 0;
-            $data->urlblocked = $urlblocked ? 1 : 0;
-            $data->timeupdated = time();
-            $DB->update_record('tool_curlmanager', $data);
+            $plugin = $this->getcomponentbycodepath($rootcodepath);
+            if ($plugin === false) {
+                $plugin = '';
+            }
 
-        } else {
-            $data = new \stdClass();
-            $data->plugin = $plugin;
-            $data->codepath = $formattedbacktrace;
-            $data->url = $urlstring;
-            $data->host = $host;
-            $data->urlallowed = $returnvalue['allowed'] ? 1 : 0;
-            $data->urlblocked = $urlblocked ? 1 : 0;
-            $data->count = 1;
-            $data->timecreated = time();
-            $data->timeupdated = time();
-            $DB->insert_record('tool_curlmanager', $data);
+            // Suggest to deduplicate on host, plugin and codepath.
+            // Check if the host, plugin and codepath exists already.
+            // Add a new record if not exist.
+            // Otherwise update the reocrd with count+1 and timeupdated field.
```

Comment:
Sounds like code planning comments, perhaps this should be moved to the relevant sections they apply for.

Reply:
This was a leftover from old code, and explains the logic of the following section. Im going to leave in place for now